src/apps/products/services/product_services.py

```diff
@@ -72,23 +73,44 @@ def create_product(
     return ProductOutputSchema.from_orm(new_product)
 
 
-def get_single_product_or_inventory(
+def get_available_single_product_or_inventory(
     session: Session, product_id: str, get_inventory=False
-) -> Union[ProductOutputSchema, InventoryOutputSchema]:
+) -> Union[ProductWithoutInventoryOutputSchema, InventoryOutputSchema, RemovedProductOutputSchema]:
     if not (product_object := if_exists(Product, "id", product_id, session)):
```

Comment:
return only available single product, cuz inventory can only be get by staff and this function seems to be for casual users so inventory obtaining doesn't need to be placed here